server/routerlicious/packages/routerlicious-base/src/riddler/tenantManager.ts

```diff
@@ -92,19 +96,20 @@ export class TenantManager {
                 }
             }
         }
-        
         // If Key 1 validation fails, try with Key 2
         try {
             await this.validateTokenWithKey(tenantKeys.key2, KeyName.key2, token);
         } catch (error) {
             if (isNetworkError(error)) {
                 if (error.code === 403) {
                     // Delete key from cache on 403
-                    Lumberjack.info(`Error with key 2 token validation. Deleting key from cache.`, lumberProperties);
-                    await this.deleteKeyFromCache(tenantId).catch((err) => {
-                        Lumberjack.error(`Error deleting keys from the cache.`, lumberProperties, err);
-                    });
-                    if (bypassCache === false) {
+                    if (bypassCache && this.isCacheEnabled) {
+                        Lumberjack.info(`Error with key 2 token validation. Deleting key from cache.`, lumberProperties);
+                        await this.deleteKeyFromCache(tenantId).catch((err) => {
+                            Lumberjack.error(`Error deleting keys from the cache.`, lumberProperties, err);
+                        });
+                    }
+                    if (!bypassCache && this.isCacheEnabled) {
```

Comment:
This one also confuses me. The comment below says:
`// Assume we used a cached key, and try again by bypassing cache.`
But the condition above will only execute if `bypassCache` is `false`... Is that intended?

Reply:
Here, it is correct. An example would be: bypassCache == false. So we try with information from the cache and it fails, because we are told not to skip cache operations. Since it fails with this error and bypassCache == false, we know that we already tried with info from the cache. So under this condition, we call this.validateToken(...) again, and set bypassCache to true, because we want to get the key directly from the database this time and skip the cache, in an attempt to validate using the information from the database

Reply:
But the same condition is being used right after the other, so I am getting rid of this repetitive if statement